DEVELOP.rst

```diff
@@ -1,347 +1,322 @@
+``<module>/src/jmh/java`` and execute them.
 
-If you want to execute specific benchmarks you can use the jar::
+If you want to execute specific benchmarks you can use the jar file, like so::
 
     $ ./gradlew :sql:jmhJar
     $ java -jar sql/build/libs/crate-sql-jmh.jar <benchmarkMethodName>
 
+Results will be generated and written to ``$buildDir/reports/jmh``.
 
-Results will be generated into ``$buildDir/reports/jmh``.
+If you're writing new benchmarks, take a look at the `JMH introduction`_ and
+`JMH samples`_.
 
-If you're writing new benchmarks take a look at this `JMH introduction`_ and
-those `JMH samples`_.
-
-Preparing a new Release
+Preparing a New Release
 =======================
 
 Before creating a new distribution, a new version and tag should be created:
 
- - Update the CURRENT version in ``io.crate.Version``.
+- Update ``CURRENT`` in ``io.crate.Version``
+- Add a section for the new version in the ``CHANGES.txt`` file
+- Commit your changes with a message like "prepare release x.y.z"
+- Push to origin
+- Create a tag by running ``./devtools/create_tag.sh``
 
- - Add a note for the new version at the ``CHANGES.txt`` file.
+You can build a release tarball like so::
 
- - Commit e.g. using message 'prepare release x.x.x'.
+    $ ./gradlew release
 
- - Push to origin
+This task runs the ``distTar`` task but also checks that the output of ``git
+describe --tag`` matches the current version of CrateDB.
 
- - Create a tag using the ``create_tag.sh`` script
-   (run ``./devtools/create_tag.sh``).
+The resulting tarball and zip file will be written to the
+``./app/build/distributions`` directory.
 
-Now everything is ready for building a new distribution, either
-manually or let Jenkins_ do the job as usual :-)
+We have a Jenkins_ job that will build the tarball for you.
 
-Building a release tarball is done via the ``release`` task. This task
-actually only runs the ``distTar`` task but additionally checks that
-the output of ``git describe --tag`` matches the current version of
-Crate::
+Navigating the Code
+===================
 
- $ ./gradlew release
+Getting familiar with a foreign code base is often a daunting task.
 
-The resulting tarball and zip will reside in the folder
-``./app/build/distributions``.
+At the moment, we don't have a full guide to the whole code architecture, but
+this section should give you an idea of where to look.
 
-Troubleshooting
-===============
+When a SQL statement is sent to CrateDB, the work-flow is roughly as follows:
 
-If you just pulled some new commits using git and get strange compile errors in
-the SQL parser code it is probably necessary to re-generate the parser code as
-the grammar changed::
+- Handle the HTTP request
 
-    ./gradlew :sql-parser:compileJava
+- Parse the request body and create an SQLRequest object (see
+  ``RestSQLAction.java``)
 
-.. _Jenkins: http://jenkins-ci.org/
+- Process the SQLRequest object (see ``doExecute`` in
```

Comment:
okay. dropping for now. if we want to add something like this in the future, we can do that as separate work. but agreed, we'd need to keep it updated

I'm thinking, because this info is likely to change over time, it might make sense if these "guides to the code base" are done as blog posts or some such? we could do them every six months or so. just thinking aloud